test/mpi/Makefile.am

```diff
@@ -12,7 +12,12 @@ if HAVE_MPICC
 
 # Test application for memory hooks when running with MPI
 # (some MPIs have hooks of their own and we make sure ours still work)
-noinst_PROGRAMS       = test_memhooks shmem_pingpong
+noinst_PROGRAMS       = test_memhooks
+
+if HAVE_SHMEMCC
```

Comment:
HAVE_SHMEMCC better be outside of HAVE_MPICC condition

Reply:
Since this is MPI test in MPI folder IMO it makes much more sense to make MPI external. SHMEM test is a subset in this case

Reply:
HAVE_MPICC and HAVE_SHMEMCC are tested independently so it makes sense to not nest them. It also makes the makefile easier to read by less nesting of conditions.